sklearn/tests/test_docstring_parameters.py

```diff
@@ -160,3 +165,100 @@ def test_tabs():
+               'SelectFromModel', 'SparseCoder', 'SparseRandomProjection',
+               'SpectralBiclustering', 'StackingClassifier',
+               'StackingRegressor', 'TfidfVectorizer', 'VotingClassifier',
+               'VotingRegressor']
+    if Estimator.__name__ in IGNORED or Estimator.__name__.startswith('_'):
+        pytest.xfail(
+            reason="Classifier cannot be fit easily to test fit attributes")
+
+    est = Estimator()
+
+    if Estimator.__name__ in ['SelectKBest']:
+        est.k = 2
+
+    X_classif, y_classif = \
+        make_classification(n_samples=20, n_features=3,
+                            n_redundant=0, n_classes=2)
+
+    X_reg, y_reg = \
+        make_regression(n_samples=10, n_features=2)
+
+    # Make sure features are positive as some models need it
+    X_classif -= X_classif.min()
+    X_reg -= X_reg.min()
+
+    if is_classifier(est):
+        X, y = X_classif, y_classif
+    elif is_regressor(est):
+        X, y = X_reg, y_reg
+    else:
+        X, y = X_classif, y_classif
+
+    tags = est._get_tags()
+
+    if '1darray' in tags['X_types']:
+        X = X[:, 0]
+
+    if tags['multioutput_only']:
+        y = np.c_[y, y]
+
+    if getattr(est, '_pairwise', None):
+        X = X.dot(X.T)
+
+    if '1dlabels' in tags['X_types']:
+        est.fit(y)
+    elif '2dlabels' in tags['X_types']:
+        est.fit(np.c_[y, y])
+    else:
+        est.fit(X, y)
+
+    for attr in attributes:
+        desc = ' '.join(attr.desc).lower()
+        if 'only ' not in desc:
+            assert hasattr(est, attr.name)
+
+    IGNORED = ['HistGradientBoostingClassifier', 'HistGradientBoostingRegressor',
+               'MiniBatchKMeans']
+    if Estimator.__name__ in IGNORED:
+        pytest.xfail(
+            reason="Classifier has too many undocumented attributes.")
+
+    fit_attr = [k for k in est.__dict__.keys() if k.endswith('_')]
+    fit_attr_names = [attr.name for attr in attributes]
+    for attr in fit_attr:
+        if attr in ['X_offset_', 'X_scale_', 'fit_', 'partial_fit_', 'x_mean_',
+                    'y_mean_', 'x_std_', 'y_std_', 'dual_gap_',
+                    'base_estimator_', 'n_classes_', 'n_estimators_',
+                    'classes_', 'n_features_', 'loss_', 'do_early_stopping_',
+                    'n_samples_fit_', 'effective_metric_params_',
+                    'effective_metric_', 'tree_', 'active_', 'alphas_',
+                    'random_state_', 'exp_dirichlet_component_',
+                    'dissimilarity_matrix_', 'n_iter_', 't_',
+                    'loss_curve_', 'best_loss_', 'eps_',
+                    'class_weight_', 'fit_status_', 'shape_fit_', 'location_',
+                    'n_nonzero_coefs_', 'loss_function_', 'random_weights_',
+                    'random_offset_', 'outlier_label_', 'n_outputs_',
+                    'one_hot_encoder_']:
+            continue
+        if attr.startswith('_'):
+            continue
+        assert attr in fit_attr_names
```

Comment:
I think here it might be better to the filtered `fit_attr` (with removed private attributes and known exceptions), than do,
```py
undocumented_attrs = set(fit_attr_names).difference(fit_attr)
assert not undocumented_attrs, "Undocumented attributes: {}".format(undocumented_attrs)
```
that way all the undocumented attributes are printed at once, and the user doesn't have to iteratively run this test.